src/libcollections/btree/node.rs

```diff
@@ -1354,11 +1354,14 @@ struct MoveTraversalImpl<K, V> {
     edges: RawItems<Node<K, V>>,
 
     // For deallocation when we are done iterating.
-    ptr: *mut u8,
+    ptr: Unique<u8>,
     capacity: usize,
     is_leaf: bool
 }
 
+unsafe impl<K: Sync, V: Sync> Sync for MoveTraversalImpl<K, V> {}
+unsafe impl<K: Send, V: Send> Send for MoveTraversalImpl<K, V> {}
```

Comment:
So raw items has basically two functions: drain items out of an allocation, and push items into an allocation. No bounds checks. No lifetimes. It's like straight up `*mut [T]`/`UncheckedVec`. 

But critically it doesn't update the source node's `len` field, because `len` is shared between three "independent" arrays. It also has no lifetimes attached, so you're free to Drop it or its backing allocation. This means that if you `Send` a RawItems alone to another thread, you basically have no idea what the state of your node's memory is at that point. It's not even sound to `Sync` RawItems since it's possible to `Drop` the parent node with an outstanding RawItems.

It's very much so an internal utility to make what is basically *mut [T] manipulation a bit nicer.